### PR TITLE
chore: remove router groups, add LayoutHOC

### PR DESCRIPTION
**Pull request by DmytroHryshyn**
## What does this PR do?

Fixes # (issue)

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Tests (Unit/Integration/E2E or any other test)
- [ ] This change requires a documentation update

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/web/app/layoutHOC.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import type { LayoutProps, PageProps } from "app/_types";
+import { cookies, headers } from "next/headers";
+
+import { buildLegacyCtx } from "@lib/buildLegacyCtx";
+
+import PageWrapper from "@components/PageWrapperAppDir";
+
+type WithLayoutParams<T> = {
+  getLayout?: ((page: React.ReactElement) => JSX.Element) | null;
```
Comment (grzpab) [inline apps/web/app/layoutHOC.tsx:9]: perhaps we should always ensure `getLayout` parameter is at least `null`?

File: apps/web/app/layoutHOC.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    const props = getData ? await getData(buildLegacyCtx(h, cookies(), params)) : null;
+
+    return (
+      <PageWrapper
+        getLayout={getLayout ?? null}
+        requiresLicense={false}
+        nonce={nonce}
+        themeBasis={null}
+        {...props}>
+        {Page ? props ? <Page {...props} /> : <></> : "children" in restProps ? restProps.children : <></>}
```
Comment (grzpab) [inline apps/web/app/layoutHOC.tsx:29]: 1. are these nested ternary expressions? (hard to follow them, perhaps we should use a helper variable?)
2. can we replace `<></>` with `null`?

**Review by hbjORbj — APPROVED** [review]
